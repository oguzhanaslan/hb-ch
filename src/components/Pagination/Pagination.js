import React from 'react';
import './Pagination.scss';

class Pagination extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hrefLink: '#', //passed as a arg like href={hrefLink} for eslint
    };
  }

  createRenderedPaginate(activePage, totalPage, pageChangeFn) {
    let paginationElements = [];

    for (let i = 1; i <= totalPage; i++) {
      paginationElements.push(
        <li key={`item-${i}`} className={`${activePage === i ? 'active' : ''}`} onClick={() => pageChangeFn(i)}>
          <a href={this.state.hrefLink}>{i}</a>
        </li>
      );
    }

    return paginationElements;
  }

  render() {
    const { activePage, totalPage, pageChangeFn } = this.props;
    return (
      <ul className='pagination'>
        <li className={`${activePage === 1 ? 'disabled' : ''}`} onClick={() => pageChangeFn('prev')}>
          <a href={this.state.hrefLink}>Prev</a>
        </li>
        {this.createRenderedPaginate(activePage, totalPage, pageChangeFn)}
        <li className={`${activePage === totalPage ? 'disabled' : ''}`} onClick={() => pageChangeFn('next')}>
          <a href={this.state.hrefLink}>Next</a>
        </li>
      </ul>
    );
  }
}

export default Pagination;
