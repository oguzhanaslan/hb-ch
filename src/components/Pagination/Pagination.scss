.pagination {
  display: flex;
  margin-bottom: 50px;
  li {
    list-style-type: none;
    flex: 1;
    margin-right: 10px;

    &.disabled {
      color: currentColor;
      cursor: not-allowed;
      opacity: 0.2;
      text-decoration: none;
      a {
        cursor: not-allowed;
      }
    }

    &:not([class*="active"]) a:hover {
      background-color: rgba(255, 255, 255, .5);
    }
    &.active a {
      background-color: #000;
      color: #fff;
      font-weight: bold;
    }

    a {
      display: flex;
      justify-content: center;
      align-items: center;
      background: #fff;
      padding: 20px 0;
      font-size: 18px;
      color: #333;
      text-decoration: none;
    }
  }
}
