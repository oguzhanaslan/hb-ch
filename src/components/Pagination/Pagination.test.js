import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';

describe('Pagination', () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<Pagination />)));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
});
