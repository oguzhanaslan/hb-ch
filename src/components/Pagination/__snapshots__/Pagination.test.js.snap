// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className=""
          onClick={[Function]}
        >
          <a
            href="#"
          >
            Prev
          </a>
        </li>,
        Array [],
        <li
          className="disabled"
          onClick={[Function]}
        >
          <a
            href="#"
          >
            Next
          </a>
        </li>,
      ],
      "className": "pagination",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
          >
            Prev
          </a>,
          "className": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Prev",
            "href": "#",
          },
          "ref": null,
          "rendered": "Prev",
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
          >
            Next
          </a>,
          "className": "disabled",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "href": "#",
          },
          "ref": null,
          "rendered": "Next",
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className=""
            onClick={[Function]}
          >
            <a
              href="#"
            >
              Prev
            </a>
          </li>,
          Array [],
          <li
            className="disabled"
            onClick={[Function]}
          >
            <a
              href="#"
            >
              Next
            </a>
          </li>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
            >
              Prev
            </a>,
            "className": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Prev",
              "href": "#",
            },
            "ref": null,
            "rendered": "Prev",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
            >
              Next
            </a>,
            "className": "disabled",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "href": "#",
            },
            "ref": null,
            "rendered": "Next",
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
